syncer/syncer.go

```diff
@@ -558,12 +558,12 @@ func (s *Syncer) syncLoop(ctx context.Context) error {
 		s.mu.Lock()
 		defer s.mu.Unlock()
 		for _, p := range s.peers {
-			if p.Synced() {
-				continue
-			}
 			peers = append(peers, p)
 		}
 		sort.Slice(peers, func(i, j int) bool {
+			if peers[i].Synced() != peers[j].Synced() {
```

Comment:
@lukechampine this seems to be one possible solution to fix the test but I'd like to hear your input on it first because it doesn't feel like a very clean one.